std/experimental/allocator/package.d

```diff
@@ -748,6 +748,15 @@ unittest
 }
 
 /// Ditto
+Unqual!(ElementType!R)[] makeArray(Allocator, R)(auto ref Allocator alloc, R range)
+if (isInputRange!R)
+{
+    import std.range.primitives;
+    alias T = Unqual!(ElementType!R);
```

Comment:
[ElementEncodingType](https://github.com/dlang/phobos/blob/v2.071.0/std/range/primitives.d#L1183):

```d
    static if (is(StringTypeOf!R) && is(R : E[], E))
        alias ElementEncodingType = E;
    else
        alias ElementEncodingType = ElementType!R
```

[ElementType](https://github.com/dlang/phobos/blob/v2.071.0/std/range/primitives.d#L1097):

```d
    static if (is(typeof(R.init.front.init) T))
        alias ElementType = T;
    else
        alias ElementType = void;
```

... where is the difference?